core/contextmenu_items.ts

```diff
@@ -331,9 +331,10 @@ export function registerDuplicate() {
       return 'hidden';
     },
     callback(scope: Scope) {
-      if (scope.block) {
-        clipboard.duplicate(scope.block);
-      }
+      if (!scope.block) return;
+      const data = scope.block.toCopyData();
```

Comment:
small suggestion: `const data = scope.block?.toCopyData()` just to minimize the number of early returns. assuming typescript is smart enough to figure out that `scope.block` must still be defined in the paste call.

Reply:
So the `?.` doesn't work in this context :/

The `?.` operator short circuits an expression to evaluate to `undefined` instead of throwing a `TypeError` because you're trying to read a property of `undefined` or `null`. It doesn't actually assert that `scope.block` is not `undefined` or `null`.